protocols/xmp/src/main/java/org/opennms/netmgt/protocols/xmp/detector/XmpDetector.java

```diff
@@ -0,0 +1,254 @@
+/*
+* OCA CONTRIBUTION ACKNOWLEDGEMENT - NOT PART OF LEGAL BOILERPLATE
+* DO NOT DUPLICATE THIS COMMENT BLOCK WHEN CREATING NEW FILES!
+*
+* This file was contributed to the OpenNMS(R) project under the
+* terms of the OpenNMS Contributor Agreement (OCA).  For details on
+* the OCA, see http://www.opennms.org/index.php/Contributor_Agreement
+*
+* Contributed under the terms of the OCA by:
+*
+* Bobby Krupczak <rdk@krupczak.org>
+* THE KRUPCZAK ORGANIZATION, LLC
+* http://www.krupczak.org/
+*/
+
+/**
+ * OpenNMS XMP Detector allows for discovery of service/protocols via 
+ * provisiond.  Our detector is pretty simple right now.  All it does is
+ * attempt to establish an XmpSession with a system and if it succeeds,
+ * it queries a few core MIB variables and returns success.
+ * Future enhancements could include more capability information like
+ * supported MIBs, modules/plugins, and various dependencies to
+ * determine if something is a server for some service (noted by an
+ * inbound dependency relationship).
+ */
+
+package org.opennms.protocols.xmp.detector;
+
+import java.net.InetAddress;
+import java.util.Date;
+import org.opennms.netmgt.provision.support.BasicDetector;
+import org.opennms.netmgt.provision.support.AbstractDetector;
+import org.opennms.netmgt.provision.ServiceDetector;
+import org.opennms.netmgt.provision.SyncServiceDetector;
+import org.krupczak.xmp.*;
+import org.opennms.netmgt.config.xmpConfig.XmpConfig;    
+import org.opennms.netmgt.protocols.xmp.config.XmpConfigFactory;
+import org.opennms.netmgt.protocols.xmp.collector.XmpCollector;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.annotation.Scope;
+import org.springframework.stereotype.Component;
+
+@Component
+/**
+ * XmpDetector class
+ * @author rdk <rdk@krupczak.org>
+ * @version $Id: $
+ *
+ */
+@Scope("prototype")
+public class XmpDetector implements SyncServiceDetector
+{
+    // class variables
```

Comment:
Please do not add '// class variables' or '// instance variables'.
The meaning is clear due to the fact that it is static.